website/project/views/tag.py

```diff
@@ -48,9 +48,37 @@ def project_add_tag(auth, node, **kwargs):
 @must_have_permission('write')
 @must_not_be_registration
 def project_remove_tag(auth, node, **kwargs):
-
     data = request.get_json()
     tag = data['tag']
     if tag:
         node.remove_tag(tag=tag, auth=auth)
         return {'status': 'success'}
+
+
+@must_be_valid_project  # injects project
+@must_have_permission('write')
+@must_not_be_registration
+def file_add_tag(auth, node, guid, **kwargs):
+    data = request.get_json()
+    tag = data['tag']
+    file_name = data['fileName']
+    if tag:
+        try:
+            fileobject = Guid.load(guid).referent
```

Comment:
This line should not be inside the `try` block - it should not raise a `ValidationError`.

I don't think it *can* raise a `ValidationError`, but if it could it might be on load if the validators applied in the field definition failed upon loading it from the database. If that happened, then you wouldn't want it to be hidden behind the behavior for the expected case of a tag's validation failing.